internal/commands/vaultsecrets/apps/read_test.go

```diff
@@ -0,0 +1,162 @@
+			},
+		},
+		{
+			Name: "No args",
+			Profile: func(t *testing.T) *profile.Profile {
+				return profile.TestProfile(t).SetOrgID("123").SetProjectID("abc")
+			},
+			Args:  []string{},
+			Error: "accepts 1 arg(s), received 0",
+		},
+		{
+			Name: "Too many args",
+			Profile: func(t *testing.T) *profile.Profile {
+				return profile.TestProfile(t).SetOrgID("123").SetProjectID("abc")
+			},
+			Args:  []string{"company-card", "additional-arg"},
+			Error: "ERROR: accepts 1 arg(s), received 2",
+		},
+	}
+	for _, c := range cases {
+		c := c
+		t.Run(c.Name, func(t *testing.T) {
+			t.Parallel()
+
+			r := require.New(t)
+			io := iostreams.Test()
+
+			ctx := &cmd.Context{
+				IO:          io,
+				Profile:     c.Profile(t),
+				ShutdownCtx: context.Background(),
+				HCP:         &client.Runtime{},
+				Output:      format.New(io),
+			}
+
+			var readOpts *ReadOpts
+			readCmd := NewCmdRead(ctx, func(o *ReadOpts) error {
+				readOpts = o
+				return nil
+			})
+			readCmd.SetIO(io)
+
+			code := readCmd.Run(c.Args)
+			if c.Error != "" {
+				r.NotZero(code)
+				r.Contains(io.Error.String(), c.Error)
+				return
+			}
+
+			r.Zero(code, io.Error.String())
+			r.NotNil(readOpts)
+			r.Equal(c.Expect.AppName, readOpts.AppName)
+			r.Equal(c.Expect.Description, readOpts.Description)
+		})
+	}
+}
+
+func TestReadRun(t *testing.T) {
+	t.Parallel()
+
+	cases := []struct {
+		Name    string
+		ErrMsg  string
+		AppName string
+	}{
+		{
+			Name:   "Failed: App not found",
+			ErrMsg: "[GET /secrets/2023-06-13/organizations/{location.organization_id}/projects/{location.project_id}/apps/{name}][403] GetApp",
+		},
+		{
+			Name:    "Success: Read app",
+			AppName: "company-card",
+		},
+	}
+
+	for _, c := range cases {
+		c := c
+		t.Run(c.Name, func(t *testing.T) {
+			t.Parallel()
+			r := require.New(t)
+
+			io := iostreams.Test()
+			vs := mock_secret_service.NewMockClientService(t)
+
+			opts := &ReadOpts{
+				Ctx:     context.Background(),
+				Profile: profile.TestProfile(t).SetOrgID("123").SetProjectID("abc"),
+				IO:      io,
+				Client:  vs,
+				Output:  format.New(io),
+				AppName: c.AppName,
+			}
+
+			if c.ErrMsg != "" {
+				vs.EXPECT().GetApp(mock.Anything, mock.Anything).Return(nil, errors.New(c.ErrMsg)).Once()
+			} else {
+				vs.EXPECT().GetApp(&secret_service.GetAppParams{
+					LocationOrganizationID: "123",
+					LocationProjectID:      "abc",
+					Name:                   opts.AppName,
+					Context:                opts.Ctx,
+				}, nil).Return(&secret_service.GetAppOK{
+					Payload: &models.Secrets20230613GetAppResponse{
+						App: &models.Secrets20230613App{
+							Name:        opts.AppName,
+							Description: opts.Description,
+						},
+					},
+				}, nil).Once()
+			}
+
+			// Run the command
+			err := readRun(opts)
+			if c.ErrMsg != "" {
+				r.Contains(err.Error(), c.ErrMsg)
+				return
+			}
+
+			r.NoError(err)
+			r.Contains(io.Output.String(), fmt.Sprintf("App Name:    %s", opts.AppName))
```

Comment:
should we keep the default format as table?

Reply:
I don't have a strong opinion. This is how table format would look.
![Screenshot 2024-05-14 at 10 13 51 AM](https://github.com/hashicorp/hcp/assets/19840012/ccdb8912-7435-4423-85ed-5776758a3f58)

Reply:
hmmn, this is the read command. I am seeing create in the pic.

Reply:
Yeah, the screenshot is old, just wanted to show how the table output looks

Reply:
Please use table here as default as in RFC we are saying tabular for read commands.